src/coreclr/nativeaot/Runtime/riscv64/WriteBarriers.S

```diff
@@ -220,31 +210,30 @@ LEAF_END RhpByRefAssignRef, _TEXT
 // on the managed heap.
 //
 // On entry:
-//   t5 : the destination address (LHS of the assignment).
+//   t3 : the destination address (LHS of the assignment).
 //         May not be a heap location (hence the checked).
-//   t6 : the object reference (RHS of the assignment).
+//   t4 : the object reference (RHS of the assignment).
 //
 // On exit:
-//   t3, t4 : trashed
-//   t5      : incremented by 8
+//   t2, t6 : trashed
+//   t3      : incremented by 8
```

Comment:
```suggestion
```
These increments are only part of `JIT_ByRefWriteBarrier` contracts.

These increments are not part of `RhpCheckedAssignRef` and `RhpAssignRef` contracts. They can be deleted here and in a few places that have this copy&paste bug.

Regular CoreCLR shares the same asm code for all write barriers, and that's how we ended up with them. (Regular CoreCLR write barrier code has TODO to stop sharing the asm code to get rid of this and other minor inefficiencies.)

Reply:
Removing that line from both `RhpCheckedAssignRef` and `RhpAssignRef` crashes before reaching the user code:

```
Thread 1 "app1" received signal SIGSEGV, Segmentation fault.
0x0000002aaac7824c in S_P_CoreLib_System_Runtime_TypeCast__StelemRef_Helper (element=..., elementType=0x2aaad29790, obj=...) at /runtime/src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs:843
843	            CastResult result = s_castCache.TryGet((nuint)obj.GetMethodTable() + (int)AssignmentVariation.BoxedSource, (nuint)elementType);
```

Keeping it reaches Console.WriteLine in Program.cs line 2 and we fail later in lock primitives https://github.com/dotnet/runtime/pull/111735#issuecomment-2639308491 (I'm working on it in separate branch).

Reply:
Ah ok, I have missed that the write barrier implementation is reused in similar way for some arches on native AOT too. We can look into this separately.